### PR TITLE
[WordPress → Jetpack app branding] Update 2FA confirmation screen

### PR DESCRIPTION
**Pull request by SiobhyB**
> **Warning**
> This PR should not be merged until D94684-code has been deployed

_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

* This PR mentions of the WordPress app on the 2FA confirmation page with references to the Jetpack app. As part of this, the page's animation has been updated with the Jetpack logo and different colours.   
* The update animation can be played below:

https://user-images.githubusercontent.com/2998162/206732827-9d81bd4d-4275-4b22-aa4c-d4852c9a40a9.mov

#### Testing Instructions

* _Prerequisite:_ A WordPress.com account with 2FA enabled as well as the Jetpack app that's logged into that account.
* With this branch checked out go through the WordPress.com login flow and view the 2FA confirmation page.
* Verify that the updated page displays as it should, with references to the Jetpack app and an updated animation.

| Before  | After |
| ------------- | ------------- |
|  |  |

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by shaunandrews** [comment]
Looking good! I did notice [the designs in Figma](https://www.figma.com/file/qk5s4q4mZsXknwhMWo2vIY/WordPress--%3E-Jetpack-Migration?node-id=7215%3A19716&t=V8brkAYKAnlyChOf-4) adjust the lines. Should we update to use the two rounded rectangles? (@osullivanchris, I think you worked on this)

CleanShot 2022-12-06 at 12 03 32@2x

--

Can we adjust the timing of the transition? I think its linear right now, but a curve (like `cubic-bezier(0.86,0,0.07,1);`) might feel a little nicer.

**Comment by osullivanchris** [comment]
> Looking good! I did notice [the designs in Figma](https://www.figma.com/file/qk5s4q4mZsXknwhMWo2vIY/WordPress--%3E-Jetpack-Migration?node-id=7215%3A19716&t=V8brkAYKAnlyChOf-4) adjust the lines. Should we update to use the two rounded rectangles? (@osullivanchris, I think you worked on this)

Yep I think the tidier lines are much better. Suits the illustration style better.

> Can we adjust the timing of the transition? I think its linear right now, but a curve (like cubic-bezier(0.86,0,0.07,1);) might feel a little nicer.

Agree I was just playing with this too. New to animation on Figma so can't achieve exactly what I wanted, but was even thinking something a bit playful and OTT like a spring or bounce could be fun, given its an illustration and not a real representation.

https://user-images.githubusercontent.com/50576199/206143500-66303bfb-4d83-4f29-847f-65dc25da029b.mov

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (fluiddot) [inline client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: @SiobhyB In theory, we can deploy these changes without a feature flag, the only requirement is to hold off the deployment until we update the 2FA push notification logic in WPCOM. Therefore, either we create another feature flag only for this change or simply hold off the merge.

**Comment by simison** [comment]
Nice to see this updated!

If the new copy is ready, recommend to add "string freeze" label to avoid shipping English-only PR.

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (fluiddot) [inline client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: I noticed that the animation is associated with the [`two-factor-authentication__illustration-notification` CSS class](https://github.com/Automattic/wp-calypso/blob/99f14b5c90c45e03d16abd2ad9598dcf0e2504a8/client/blocks/login/two-factor-authentication/push-notification-illustration.scss#L38-L52), in order to animate the Jetpack icon version, we should include it here.

**UPDATE:** Probably we don't need the `xmlns`, `fill` and `viewBox` properties, based on the previous SVG.

**Review by fluiddot — COMMENTED** [review]
@SiobhyB Thanks for making this update to the 2FA confirmation screen 🙇 ! 

I tested the changes and look great. However, I couldn't reproduce the animation in the notification component as described in the PR. In relation to this, I added a comment with a potential solution.

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (SiobhyB) [reply client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: Thanks for the clarification! I've gone ahead to remove this flag in https://github.com/Automattic/wp-calypso/pull/70832/commits/5a9ebfb06eaca32bb72f2d259526c1cb33011c61.

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (SiobhyB) [reply client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: Ah, apologies, I think you caught me mid-update! You should be able to view the animation as expected with the latest changes. 🙇‍♀️

**Comment by SiobhyB** [comment]
@shaunandrews, @osullivanchris, thanks for the feedback! I've updated the lines in the illustration and also experimented a bit with the animation. Let me know what you think of this, happy to tweak further depending on your thoughts:

https://user-images.githubusercontent.com/2998162/206278989-5027ccb7-c3e1-43b5-9eb0-8dc9153a7f02.mov

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7662290

Thank you @SiobhyB for including a screenshot in the description! This is really helpful for our translators.

**Comment by shaunandrews** [comment]
Manually doing the bounce with keyframes doesn't work very well; It feels clunky, and the keyframes are too obvious in the resulting visual.

If there was a way you could do it with [EaseOutBounce](https://easings.net/#easeOutBounce) that would probably feel a lot better. I think if you go back to just the two keyframes, and adjust the animation-timing it might work. Or, you could probably do it with `transition` and `transition-delay`.

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (fluiddot) [inline client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: I noticed that we include the `jetpack` CSS class. However, I couldn't find what styles are added to the element when I inspected the HTML, so I'm wondering if we need it, wdyt?

**Review by fluiddot — COMMENTED** [review]
Following [this comment](https://github.com/Automattic/wp-calypso/pull/70832#issuecomment-1341608222), I also found that the bounce animation could be improved. Probably we could apply the suggestions outlined in the comment. Apart from this, the PR is ready from my side.

**Comment by SiobhyB** [comment]
> If there was a way you could do it with [EaseOutBounce](https://easings.net/#easeOutBounce) that would probably feel a lot better.

Thanks for this tip @shaunandrews! I see that the `EaseOutBounce` function generates multiple keyframes, but obviously a lot more thought out and calculated than my manual attempt. I've gone ahead to use the keyframes from that function, which produces the following result:

https://user-images.githubusercontent.com/2998162/206694015-4db295a4-a1e3-4103-871d-2e88fd2c621a.mov

How does this seem?

File: client/blocks/login/two-factor-authentication/push-notification-illustration.jsx
Comment (SiobhyB) [reply client/blocks/login/two-factor-authentication/push-notification-illustration.jsx]: You're right, it's no longer needed, thanks for catching! Removed that in https://github.com/Automattic/wp-calypso/pull/70832/commits/53043bb141909b01b394fdca626e3a1dce9569ff.

**Comment by fluiddot** [comment]
@shaunandrews @SiobhyB I'm not sure about the bouncing effect 🤔 . Usually, notifications just slide from the top without bouncing, couldn't we use a quadratic curve like `easeOutCubic` to simply animate the vertical position?

On the other hand, this change is a bit time-sensitive as it should be aligned with the release of WordPress app version `21.3` on Dec 12th, 2022, so I wouldn't advocate delaying this PR too much due to iterating the animation.

**Comment by fluiddot** [comment]
@osullivanchris  @SiobhyB I'm not sure about the bouncing effect 🤔 . Usually, notifications just slide from the top without bouncing, couldn't we use a quadratic curve like `easeOutCubic` to simply animate the vertical position?

On the other hand, this change is a bit time-sensitive as it should be aligned with the release of WordPress app version `21.3` on Dec 12th, 2022, so I wouldn't advocate delaying this PR too much due to iterating the animation.

**Comment by SiobhyB** [comment]
As the main changes to the screen's text/icon are time sensitive, I've gone ahead to revert the bouncing effect and have left the animation as-is for now, with the suggested change to the `cubic-bezier` property:

https://user-images.githubusercontent.com/2998162/206732827-9d81bd4d-4275-4b22-aa4c-d4852c9a40a9.mov

@osullivanchris, @shaunandrews, I'll be happy to iterate on the animation in a separate PR to this one, let me know if that's something you'd like me to work on.

**Comment by osullivanchris** [comment]
👋  Yeah the bouncy one was a bit much! The latest looks better to me. The properties you mentioned I assume are the ones @shaunandrews defined, so that sounds good to me. Will let him confirm.

**Comment by shaunandrews** [comment]
The animation looks good to me. 🚢

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I noticed that one check is failing (_Odyssey Stats (WPCom Plugins)_), however it doesn't look like it's related to these changes. Maybe rebasing this PR with `trunk` would solve it.

## IMPORTANT NOTE
**This PR should only be merged once this D94684-code diff is merged in WPCOM.** 

@SiobhyB maybe we should add a note in the PR's description with this information to warn other reviewers about this.

**Comment by SiobhyB** [comment]
Thanks for the review @fluiddot! 🙌 I've gone ahead to add a warning to the PR description, as per your advice :)

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7662290).

**Pull request merged**
